src/models/albert/encoder.rs

```diff
@@ -0,0 +1,375 @@
+        let n_hidden_groups = hf_config.num_hidden_groups;
+        let n_hidden_layers = hf_config.num_hidden_layers;
+
+        ensure!(
+            n_hidden_layers % n_hidden_groups == 0,
+            IncorrectNHiddenGroupsSnafu {
+                n_hidden_groups,
+                n_hidden_layers
+            }
+        );
+
+        let attention_probs_dropout =
+            Box::new(DropoutConfig::default().p(hf_config.attention_probs_dropout_prob));
+        let hidden_dropout = Box::new(DropoutConfig::default().p(hf_config.hidden_dropout_prob));
+        let embedding_layer_norm = Box::new(
+            LayerNormConfig::default()
+                .eps(hf_config.layer_norm_eps)
+                .size(hf_config.embedding_size),
+        );
+        let layer_norm = Box::new(
+            LayerNormConfig::default()
+                .eps(hf_config.layer_norm_eps)
+                .size(hf_config.hidden_size),
+        );
+
+        let embeddings = TransformerEmbeddingsConfig::default()
+            .embedding_dropout(hidden_dropout.clone())
+            .embedding_layer_norm(embedding_layer_norm)
+            .embedding_width(hf_config.embedding_size)
+            .hidden_width(hf_config.hidden_size)
+            .n_pieces(hf_config.vocab_size)
+            .n_positions(Some(hf_config.max_position_embeddings))
+            .n_types(Some(hf_config.type_vocab_size));
+
+        let attention = SelfAttentionConfig::default()
+            .attention_heads(AttentionHeads {
+                n_query_heads: hf_config.num_attention_heads,
+                n_key_value_heads: hf_config.num_attention_heads,
+                qkv_mode: QkvMode::Separate,
+            })
+            .attention_scorer(Box::new(
+                ScaledDotProductAttentionConfig::default().dropout(attention_probs_dropout),
+            ))
+            .hidden_width(hf_config.hidden_size);
+
+        let feedforward = PointwiseFeedForwardConfig::default()
+            .activation(Box::new(hf_config.hidden_act))
+            .dropout(hidden_dropout)
+            .hidden_width(hf_config.hidden_size)
+            .intermediate_width(hf_config.intermediate_size);
+
+        let layer = AlbertLayerGroupConfig::default()
+            .n_layers_per_group(hf_config.inner_group_num)
+            .transformer_layer(
+                TransformerLayerConfig::default()
+                    .attention(attention)
+                    .attn_residual_layer_norm(layer_norm.clone())
+                    .feedforward(feedforward)
+                    .ffn_residual_layer_norm(layer_norm),
+            );
+
+        Ok(Self::default()
+            .embeddings(Box::new(embeddings))
+            .layer_group(Box::new(layer))
+            .n_hidden_groups(n_hidden_groups)
+            .n_hidden_layers(n_hidden_layers))
+    }
+}
+
+/// ALBERT encoder configuration.
+///
+/// Configuration for [AlbertEncoder]. ALBERT differs from an encoder like
```

Comment:
Are the brackets intentional?

Reply:
Yes, generates a link to `AlbertEncoder` in the documentation.